plan/rule_partition_processor.go

```diff
@@ -105,6 +105,12 @@ func (s *partitionProcessor) prune(ds *DataSource) (LogicalPlan, error) {
 		newDataSource.baseLogicalPlan = newBaseLogicalPlan(ds.context(), TypeTableScan, &newDataSource)
 		newDataSource.isPartition = true
 		newDataSource.partitionID = pi.Definitions[i].ID
+		// TODO: Find a better way to handle here.
```

Comment:
I think we can remove this TODO

Reply:
Any suggestion?

Reply:
I think if we haven't find a way to optimize the old logic, we don't need a TODO.